packages/eslint-plugin-query/src/rules/exhaustive-deps/exhaustive-deps.rule.ts

```
@@ -182,3 +164,27 @@ function getQueryFnRelevantNode(queryFn: TSESTree.Property) {
+
+function dereferenceVariablesAndTypeAssertions(queryKeyNode: TSESTree.Node, context: Readonly<TSESLint.RuleContext<string, ReadonlyArray<unknown>>>) {
+  for (let i = 0; i < (1 << 16); ++i) {
+    switch (queryKeyNode.type) {
+      case AST_NODE_TYPES.TSAsExpression:
+        queryKeyNode = queryKeyNode.expression
+        break
+      case AST_NODE_TYPES.Identifier:
+        const expression = ASTUtils.getReferencedExpressionByIdentifier({
+          context,
+          node: queryKeyNode,
+        })
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)